[17.0][MIG/IMP] sale_validity_auto_cancel: Migration to 17.0

fd-oerp opened the pull request:
This module has been migrated to V17.0, and a new functionality has been added.

Description:

The 'Keep SO' field is a boolean field added to the res.partner model.
This field is connected to the sale.order model using a related field.

Functionality/Use:
This feature provides a dynamic configuration that allows users to manage sales orders (SOs) based on the partner ID. If the 'Keep SO' field is checked, it prevents the automatic cancellation of sales orders associated with that partner.

With this functionality, clients can enforce specific restrictions. For instance, certain sales orders can be protected from being canceled if they are linked to partners for whom the 'Keep SO' field is enabled.

This will enhances operational efficiency by ensuring critical sales orders remain active and are not canceled inadvertently. This new feature aims to improve the management of sales orders, providing greater customization and control over the sales process.

JordiMForgeFlow committed:
sale_validity_auto_cancel

Commit:
[IMP] sale_validity_auto_cancel: black, isort, prettier

Commit:
[MIG] sale_valitidy_auto_cancel: migration to 14.0

Commit:
[UPD] Update sale_validity_auto_cancel.pot

OCA-git-bot committed:
[BOT] post-merge updates

fd-oerp committed:
[IMP] sale_validity_auto_cancel: pre-commit auto fixes

fd-oerp commented:
Hi @JordiMForgeFlow 

Could you please review this PR and let me know if anything needs to be updated from my end?

Thank you

JordiMForgeFlow commented on sale_validity_auto_cancel/models/res_partner.py:
```diff
@@ -0,0 +1,11 @@
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    is_keep_so = fields.Boolean(string="Keep Expired SO")
```
I would recommend a more descriptive technical name for the field taking into account the module naming. For example it could be like 'auto_cancel_expired_so', which would be True for all partners by default.

fd-oerp replied:
Hi @JordiMForgeFlow 

Thank you for your reply and suggestion, I'll update accordingly and will request for review.

Thank you

JordiMForgeFlow reviewed (commented):
It would be best to split the commits: 1 commit for the MIG and one for the IMP to add the partner field :)

rousseldenis commented:
/ocabot migration sale_validity_auto_cancel

rousseldenis commented on sale_validity_auto_cancel/views/res_partner_views.xml:
```diff
@@ -0,0 +1,20 @@
+  License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+-->
+<odoo>
+    <data>
```
You can remove this tag

rousseldenis reviewed (commented):
Code review.

@fd-oerp Could you update this according to @JordiMForgeFlow comment ?

fd-oerp committed:
[17.0][MIG] sale_validity_auto_cancel: Migration to 17.0

fd-oerp committed:
[17.0][IMP] sale_validity_auto_cancel: Extend the auto cancel expired SO functionality by adding new field to contact
[17.0][IMP] sale_validity_auto_cancel: Extend the auto cancel expired SO functionality by adding new field to contact

[17.0][IMP] sale_validity_auto_cancel: Extend the auto cancel expired SO functionality by adding new field to contact

fd-oerp commented:
Hi @JordiMForgeFlow @rousseldenis 

I have updated the commits as you suggested, with one for MIG and one for IMP. Additionally, I have updated the field name to be suitable according to the module's technical name.

I'm not sure why one check failed—it seems like it might be due to another module. Could you please guide me if I'm missing something related to the failed check?

Please review and let me know if anything needs to be updated from my end.

Thank you!

JordiMForgeFlow reviewed (approved):
code review 👍🏼

fd-oerp commented:
Hi @rousseldenis 

Could you please review the changes I made as per @JordiMForgeFlow's suggestions and proceed with this PR accordingly? Let me know if anything needs to be updated from my end.

Thank you

Deriman-Alonso reviewed (approved):
Code review, seems fine.

fd-oerp commented:
Hi,

Could this PR proceed further, or please let me know if anything needs to be updated on my end?

Thank you

fd-oerp commented:
Hi @rousseldenis 

Could this PR proceed further, or please let me know if anything needs to be updated on my end?

Thank you

Jaimermaccione reviewed (approved):
LGTM! Functional review on local enviroment

manuelregidor reviewed (approved):
Technical review. LGTM

HaraldPanten commented:
/ocabot merge nobump
